packages/storybook/stories/components/combobox/documentation.mdx

```diff
@@ -4,50 +4,170 @@ import { Banner } from '../../../src/components/banner/Banner';
+3. <StorybookLink kind={ ODS_COMPONENTS_TITLE.tag } label="Tag" story={ STORY.documentation } /> _(multiple selection mode)_ to display selected items as tags inside the Input field
+4. **Clearable** <StorybookLink kind={ ODS_COMPONENTS_TITLE.button } label="Button" story={ STORY.documentation } /> _(optional)_ to allow users to clear the input content
+5. <StorybookLink kind={ ODS_COMPONENTS_TITLE.spinner } label="Spinner" story={ STORY.documentation } /> _(optional)_ to indicate that data is being fetched
+6. **Add entry option** _(optional)_ allows users to create new entries when no matching result is found. The label is customizable
+7. **Empty state message** is a customizable message displayed when no suggestion match the query
 
 <Heading label="Usage" level={ 2 } />
 
-TODO
+The **Combobox** is best suited when users need to:
+* search within a dataset and dynamically refine results
+* provide suggestions based on user input (e.g., domain names, tags, predictive search)
+* allow users to add custom values when applicable (e.g., creating tags)
 
 <Heading label="Dos & Don'ts" level={ 3 } />
 
 <BestPractices
-  donts={[]}
-  dos={[]}
+  donts={[
+    '- Avoid excessive grouping of items, as too many categories can overwhelm users',
+    '- Avoid using Combobox when the number of items is very small, you can use Select or Radio',
+  ]}
+  dos={[
+    '- Use Combobox for datasets where typing helps filter results',
+    '- Suitable for datasets up to a few hundred entries',
+    '- Provide meaningful empty states',
+    '- Allow users to add custom entries when appropriate',
+  ]}
 />
 
 <Heading label="Placement" level={ 2 } />
 
-TODO
+The dropdown is positioned below the Input field when there is enough space.
+
+The dropdown width should match the Input field width.
+
+In multiple selection mode, the Input field height grows dynamically to accommodate selected tags.
 
 <Heading label="Behavior" level={ 2 } />
 
-TODO
+<Heading label="Triggering the dropdown" level={ 3 } />
+
+The dropdown appears when the user clicks on the input field.
+
+<Heading label="Selecting items" level={ 3 } />
+
+Selecting an item triggers a custom event, allowing integrators to process the selected value(s).
+
+<Heading label="Single selection mode" level={ 4 } />
+
+Clicking on an item selects it, closes the dropdown, and updates the Input field value.
+
+If the user exists the field without selecting an item, the input reverts to the placeholder or the last selected value (if any).
+
+Users cannot select the same item twice; already selected items are removed from the dropdown.
+
+<Heading label="Multiple selection mode" level={ 4 } />
+
+Clicking on an item add it as a tag inside the Input field.
+
+Selected items do not reappear in the dropdown to avoid duplicates.
+
+Clicking on a tag removes it.
+
+If a previously added entry tag is removed, its corresponding item does not reappear in the dropdown.
```

Comment:
ambiguous I guess you mean entry that were not in the original list, but seems unclear especially as you describe this feature just after